api/src/main/java/org/openmrs/api/db/hibernate/HibernatePersonDAO.java

```diff
@@ -202,23 +226,7 @@ public void setSessionFactory(SessionFactory sessionFactory) {
 		} else if (names.length == 2) {
```

Comment:
Some comments on this function now:

* We seem to have eliminated the need for the `result` variable at all, so that should probably be removed.
* The line `StringBuilder q = ...` should be moved to the final `else` block, since that's the only place it's necessary.
* Either the final `else` block should be removed or everything after the end of the final `else` block should be added into the final `else` block as that common logic only applies when we continue to support a case for multiple names.

Reply:
Make sense! I am planing to also refactor the else block. But for nut it should be good to move everything in it!

Reply:
Fixed all these issues in the new commit!